### PR TITLE
Improvements 1a

### PR DESCRIPTION
**Pull request by edbeeching**
This PR:
- [x] Cleans up setup.py and remove settings.ini (legacy from nbdev)
- [x] Makes all internal imports relative (e.g. from trl.trainer -> from .trainer)
- [x] Cleans up README and fixes the example

### CODE REVIEW

File: README.md
Comment (lvwerra) [inline README.md:58]: we can now use:

```suggestion
gpt2_model_ref = create_refrence_model(gpt2_model)
```

or we can not provide a reference model at all in which case the trainer will call above functions.

File: README.md
Comment (lvwerra) [inline README.md:72]: Why did you change this? I think the response should be given by the active model, no?

File: README.md
Comment (lvwerra) [inline README.md:87]: This seems unnecessary?

File: README.md
Comment (lvwerra) [inline README.md:57]: What do you think about omitting `gpt2_` from tokenizer and model names?

File: README.md
Comment (lvwerra) [inline README.md:102]: We should let `accelerate` handle device placement internally. (Can address it in a follow up PR if you want)

File: setup.py
Comment (lvwerra) [inline setup.py:27]: Since we don't test it anymore we could also remove it here, right?
```suggestion
```

File: trl/trainer/__init__.py
Comment (lvwerra) [inline trl/trainer/__init__.py:21]: Why are the utils imports separate from the others?

**Review by lvwerra — COMMENTED** [review]
Thanks @edbeeching, this looks great! Left a few minor comments, can also address some of them in a follow up PR.

File: README.md
Comment (edbeeching) [reply README.md:72]: Yes you are right, in this case the model is just used to create the data for the dummy dataset, but I will update it because in a normal training setting it would not be like this.

File: README.md
Comment (edbeeching) [reply README.md:102]: I think the reward would normally be from a preference model, so it would probably be on the correct device. But in this example that is not the case. I'll leave it to another PR.

Comment (lvwerra) [reply README.md:100]: yes, discussed with @younesbelkada that we add some device placement steps to `PPOTrainer.step`.

**Review by lvwerra — APPROVED** [review]
Looks good - thanks @edbeeching!

**Review by younesbelkada — APPROVED** [review]
Thanks for taking care of the improvements @edbeeching !

File: README.md
Comment (younesbelkada) [reply README.md:100]: yes, this will be addressed in a PR right after merging this

**Pull request merged**
